saos-search/src/main/java/pl/edu/icm/saos/search/indexing/ReindexJobStepExecutionListener.java

```diff
@@ -35,6 +44,14 @@
     public void beforeStep(StepExecution stepExecution) {
         SourceCode sc = EnumUtils.getEnum(SourceCode.class, sourceCode);
         judgmentRepository.markAsNotIndexedBySourceCode(sc);
+        
+        if (sc == null) {
```

Comment:
Krzysiek, now I think it would be great to execute the deleteByQuery even if sc!=null with slightly different another delete query that deletes only the judgments with the specifed sourceCode. 
It would be something like that:
    String deletingQuery = "*";
    if (sc != null) {
           deletingQuery = ...
    }
    try {
       solrJudgmentsServer.deleteByQuery(deletingQuery);
    }


      }
    } catch...